file.go

```diff
@@ -1,37 +1,121 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
 
 type file struct {
 	// TODO: define variables if needed
+	JSONFormat map[string]interface{}
+	CSVFormat  struct {
+		Data [][]string
+		//if "headers==true", data[0] will contain headers
+		HasHeaders bool
+	}
 }
 
+func (f *file) initialize() {
+	f.JSONFormat = nil
+	f.CSVFormat.Data = nil
+	f.CSVFormat.HasHeaders = false
+}
 func ParseJsonFile(path string) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	jsonFile, err := os.Open(path)
+	var myFile file
+	myFile.initialize()
+	if err != nil {
+		return myFile, err
+	}
+	defer jsonFile.Close()
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err == nil {
+		json.Unmarshal([]byte(byteValue), &myFile.JSONFormat)
+	}
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
 	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var myFile file
+	myFile.initialize()
+	byteValue, err := ioutil.ReadAll(f)
+	if err == nil {
+		json.Unmarshal([]byte(byteValue), &myFile.JSONFormat)
+	}
+	return myFile, err
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	csvFile, err := os.Open((path))
+	var myFile file
+	myFile.initialize()
+	if err != nil {
+		return myFile, err
+	}
+	defer csvFile.Close()
+	csvReader := csv.NewReader(csvFile)
+	myFile.CSVFormat.Data, err = csvReader.ReadAll()
+	if err == nil {
+		myFile.CSVFormat.HasHeaders = hasHeaders
+		myFile.JSONFormat = nil
+	}
+	return myFile, err
 }
 
-func ParseCsvJsonFile(f *os.File, hasHeaders bool) (file, error) {
+func ParseOpenedCsvFile(f *os.File, hasHeaders bool) (file, error) {
 	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var myFile file
+	myFile.initialize()
+	var err error
+	csvReader := csv.NewReader(f)
+	myFile.CSVFormat.Data, err = csvReader.ReadAll()
+	if err == nil {
+		myFile.CSVFormat.HasHeaders = hasHeaders
+		myFile.JSONFormat = nil
+	}
+	return myFile, err
 }
 
 func (f file) Csv(addHeaders bool) []byte {
 	// TODO: return the file data in CSV format
-	return nil
+	if f.CSVFormat.Data == nil {
+		return nil
+	}
```

Comment:
So I can't do this?
```
jcsv.ParseJsonFile("input.json").Csv(true)
```

What is the purpose of this package then?

Reply:
Faizan Bhai, I changed the implementation, now it is supporting the conversion from JSON to CSV and CSV to JSON. But the statement you wrote still produces a syntax error. ParseJsonFile is returning (file, err) while Csv is a method of file.
![image](https://user-images.githubusercontent.com/87894106/140386623-ad951de8-da8a-4af7-b96f-187f9a5801a6.png)

Reply:
![image](https://user-images.githubusercontent.com/87894106/140386897-8e72dff5-6fe4-402b-898e-3b6cf0d81649.png)